include/secp256k1_schnorrsig.h

```diff
@@ -23,24 +23,28 @@ extern "C" {
  *
  *  Returns: 1 if a nonce was successfully generated. 0 will cause signing to
  *           return an error.
- *  Out:     nonce32:   pointer to a 32-byte array to be filled by the function.
- *  In:      msg32:     the 32-byte message hash being verified (will not be NULL)
- *           key32:     pointer to a 32-byte secret key (will not be NULL)
- *      xonly_pk32:     the 32-byte serialized xonly pubkey corresponding to key32
- *                      (will not be NULL)
- *           algo16:    pointer to a 16-byte array describing the signature
- *                      algorithm (will not be NULL).
- *           data:      Arbitrary data pointer that is passed through.
+ *  Out:  nonce32: pointer to a 32-byte array to be filled by the function.
+ *  In:       msg: the message being verified (will not be NULL)
+ *         msglen: the length of the message (will not be NULL)
```

Comment:
```suggestion
 *  In:       msg: the message hash being verified (will not be NULL)
 *         msglen: the length of the message hash (will not be NULL)
```

Reply:
No, messages in BIP340 can be arbitrary bytestrings. They do not need to be obtained via hashing.

Reply:
Ah, right, hence this PR. Missed that somehow on my first pass.

Reply:
Ah no, I think there's still a misunderstanding. So with the scheme in the current BIP340, you can already sign arbitrary *32-byte* messages (no matter if they're created via a hash function or not). 

There have been changes proposed to BIP340 (not yet in the BIP) that extend this to arbitrary-length messages. This is what is implemented by this PR. 

But BIP340 never required the messages to be created via a hash function. This is just a difference to the ECDSA API.

Reply:
Thanks for clarifying. I'm assuming that despite the fact that BIP340, unlike ECDSA, doesn't require a hashed message, when the documentation on line 27 stated previously `the 32-byte message hash being verified` that was not an error, because for typical use cases (i.e. the message is not exactly 32-bytes) the message would need to be hashed.

However, with the introduction of variable length messages in this PR, the documentation no longer makes this assumption about the typical use case because it no longer requires a 32-byte message for the APIs that now accept a `msglen`. This seems to also be reflected in the insertion of `hash` in this PR on line 109, because that API still requires a 32-byte message. So the documentation logic of this PR seems to be that if `msglen` is specified, the documentation should describe it as simply a `message` whereas when a 32-byte message is required then the documentation should describe it as a `32-byte message hash` (even though technically a hash is not required, but for typical use cases it should be a hash because messages won't typically be strictly 32-bytes).

Reply:
I'm not sure I follow, what exactly is the possible documentation error? I agree that "message hash" for schnorrsig_sign's msg32 argument is misleading because it's not necessary that msg32 is one. Switched it back.

Reply:
The possible documentation error I was referring to is that  currently line 27 on master says: https://github.com/bitcoin-core/secp256k1/blob/3a8b47bc6d1627a8f9f008234c5161538ef85a21/include/secp256k1_schnorrsig.h#L27 when it should probably be `the 32-byte message being verified`.

With your latest changes the documentation for both `secp256k1_nonce_function_hardened` and `secp256k1_schnorrsig_sign` now have `message` instead of `message hash` so it now looks good to me.